src/lib/browse/CensusOutputAreaLayerControl.svelte

```diff
@@ -0,0 +1,244 @@
+    } else {
+      $map.setLayoutProperty(outlineLayer, "visibility", "none");
+    }
+  }
+
+  overwritePmtilesSource(
+    $map,
+    name,
+    `https://atip.uk/layers/v1/${name}.pmtiles`
+  );
+
+  overwritePolygonLayer($map, {
+    id: name,
+    source: name,
+    sourceLayer: name,
+    // Initially set to a dummy value
+    color: "black",
+    opacity: hoveredToggle(0.5, 0.7),
+  });
+  overwriteLineLayer($map, {
+    id: outlineLayer,
+    source: name,
+    sourceLayer: name,
+    color: "black",
+    width: 0.5,
+  });
+
+  function tooltip(feature: MapGeoJSONFeature): string {
+    let oa = feature.properties["OA21CD"];
+    let value = feature.properties[colorBy];
+    if (colorBy == "percent_households_without_car") {
+      return `<p>${value}% of households in ${oa} don't have a car</p>`;
+    } else if (colorBy == "average_cars_per_household") {
+      return `<p>Households in ${oa} have an average of ${value} cars</p>`;
+    } else {
+      return `<p>There are ${value.toLocaleString()} people per square kilometre in ${oa}</p>`;
+    }
+  }
+
+  function onClick(e: CustomEvent<MapGeoJSONFeature>) {
+    let oa = e.detail.properties["OA21CD"];
+    if (
+      colorBy == "percent_households_without_car" ||
+      colorBy == "average_cars_per_household"
+    ) {
+      window.open(
+        `https://www.ons.gov.uk/census/maps/choropleth/housing/number-of-cars-or-vans/number-of-cars-5a/no-cars-or-vans-in-household?oa=${oa}`,
+        "_blank"
+      );
+    } else {
+      window.open(
+        `https://www.ons.gov.uk/census/maps/choropleth/population/population-density/population-density/persons-per-square-kilometre?oa=${oa}`,
+        "_blank"
+      );
+    }
+  }
+
+  // Should return 6 things (1 more than colorScale)
+  function makeLimits(): number[] {
+    if (colorBy == "percent_households_without_car") {
+      // Equal buckets of 20%
+      return [0, 20, 40, 60, 80, 100];
+    } else if (colorBy == "average_cars_per_household") {
+      // The max value is 2.4, so pick equal buckets
+      return [0.0, 0.5, 1.0, 1.5, 2.0, 2.5];
+    } else {
+      // Use the same (slightly rounded) buckets as https://www.ons.gov.uk/census/maps/choropleth/population/population-density/population-density/persons-per-square-kilometre
+      return [0, 4700, 13000, 33000, 94000, 1980000];
+    }
+  }
+
+  function makeColorRamp(): any[] {
+    let limits = makeLimits();
+    let fillColor: any[] = ["step", ["get", colorBy]];
+    for (let i = 1; i < limits.length; i++) {
+      fillColor.push(colorScale[i - 1]);
+      fillColor.push(limits[i]);
+    }
+    // If we have data greater than the upper limit, then this fallback color
+    // is used. It's effectively a bug -- the limits need to be set correctly.
+    fillColor.push("red");
+    return fillColor;
+  }
+</script>
+
+<Checkbox
+  id="percent_households_without_car"
+  bind:checked={showHouseholdsWithoutCar}
+  on:change={() => {
```

Comment:
Manual radio button logic :)

Reply:
I've not gotten the opportunity to use this Svelte feature yet, but I wonder if https://svelte.dev/tutorial/sharing-code might be useful here?! Might just be a complicated answer to a simple problem 🙃

Reply:
Ooh, thanks, that is clever! There could be a component wrapping these checkboxes. I'm not sure what the order would be though... does `on:change` happen after the checkbox state is toggled on? And in that shared `stopOthers` method, can we detect the "current" checkbox and keep it on? Maybe...